File: src/main/java/org/carlosalcina/java/tareas/Usuario.java
package org.carlosalcina.java.tareas;

import java.util.ArrayList;
import java.util.List;

public class Usuario {
    private final String nombre;
    private final List<Proyecto> proyectos;

    public Usuario(String nombre) {
        this.nombre = nombre;
        this.proyectos = new ArrayList<>();
    }

    public void crearProyecto(String nombreProyecto) {
        Proyecto proyecto = new Proyecto(nombreProyecto);
        proyectos.add(proyecto);
        System.out.println(nombre + " ha creado el proyecto '" + nombreProyecto + "'.");
    }

    public void agregarTareaAProyecto(String nombreProyecto, Tarea tarea) {
        for (Proyecto proyecto : proyectos) {
            if (proyecto.nombre.equals(nombreProyecto)) {
                proyecto.agregarTarea(tarea);
                break;
            }
        }
    }

    public void completarTareaEnProyecto(String nombreProyecto, String nombreTarea) {
        for (Proyecto proyecto : proyectos) {
            if (proyecto.nombre.equals(nombreProyecto)) {
                proyecto.completarTarea(nombreTarea);
                break;
            }
        }
    }

    public void mostrarProyectos() {
        System.out.println("Proyectos de " + nombre + ":");
        for (Proyecto proyecto : proyectos) {
            proyecto.mostrarTareas();
        }
    }
}

File: src/main/java/org/carlosalcina/kotlin/tareas/Usuario.kt
package org.carlosalcina.kotlin.tareas

class Usuario(val nombre: String) {
    private val proyectos = mutableListOf<Proyecto>()

    fun crearProyecto(nombreProyecto: String) {
        val proyecto = Proyecto(nombreProyecto)
        proyectos.add(proyecto)
        println("$nombre ha creado el proyecto $nombreProyecto")
    }

    fun agregarTareaAProyecto(nombreProyecto: String, tarea: Tarea) {
        val proyecto = proyectos.find { it.nombre == nombreProyecto }
        proyecto?.agregarTarea(tarea)
    }

    fun completarTareaEnProyecto(nombreProyecto: String, nombreTarea: String) {
        val proyecto = proyectos.find { it.nombre == nombreProyecto }
        proyecto?.completarTarea(nombreTarea)
    }

    fun mostrarProyectos() {
        println("Proyectos de $nombre:")
        proyectos.forEach { it.mostrarTareas() }
    }
}